pkg/probe/probes.go

```diff
@@ -96,54 +118,137 @@ func checkNodeReadiness(client client.Client, timeout time.Duration) error {
 func defaultGw() (string, error) {
 	defaultGw := ""
 	return defaultGw, wait.PollImmediate(1*time.Second, defaultGwRetrieveTimeout, func() (bool, error) {
-		observedStateRaw, err := nmstatectl.Show()
+		gjsonCurrentState, err := currentStateAsGJson()
 		if err != nil {
-			log.Error(err, fmt.Sprintf("failed retrieving current state"))
-			return false, nil
+			return false, errors.Wrap(err, "failed retrieving current state to retrieve default gw")
 		}
-
-		currentState, err := yaml.YAMLToJSON([]byte(observedStateRaw))
-		if err != nil {
-			return false, errors.Wrap(err, "failed to convert current state to JSON")
-		}
-
-		defaultGw = gjson.ParseBytes(currentState).
+		defaultGw = gjsonCurrentState.
 			Get("routes.running.#(destination==\"0.0.0.0/0\").next-hop-address").String()
 		if defaultGw == "" {
-			log.Info("default gw missing", "state", string(currentState))
+			log.Info("default gw missing", "state", gjsonCurrentState.String())
 			return false, nil
 		}
 
 		return true, nil
 	})
 }
 
-func RunAll(client client.Client) error {
+func runPing(client client.Client, timeout time.Duration) error {
 	defaultGw, err := defaultGw()
 	if err != nil {
 		return errors.Wrap(err, "failed to retrieve default gw at runProbes")
 	}
 
-	currentState, err := nmstatectl.Show()
+	// TODO: Make ping timeout configurable with a config map
```

Comment:
i would open issue and not put TODO ? but you just moved it, so up to you

Reply:
Done https://github.com/nmstate/kubernetes-nmstate/issues/609